src/machine/operation.c

```diff
@@ -88,36 +88,34 @@ static int operation_done(sd_event_source *s, const siginfo_t *si, void *userdat
         return 0;
 }
 
-int operation_new(Manager *manager, Machine *machine, pid_t child, sd_bus_message *message, sd_varlink *link, int errno_fd, Operation **ret) {
+int operation_new(Manager *manager, Machine *machine, pid_t child, int errno_fd, Operation **ret) {
         Operation *o;
         int r;
 
         assert(manager);
         assert(child > 1);
         assert(errno_fd >= 0);
-        assert(message || link);
-        assert(!(message && link));
+        assert(ret);
 
-        o = new0(Operation, 1);
+        o = new(Operation, 1);
         if (!o)
                 return -ENOMEM;
 
-        o->extra_fd = -EBADF;
-
         r = sd_event_add_child(manager->event, &o->event_source, child, WEXITED, operation_done, o);
         if (r < 0) {
                 free(o);
                 return r;
         }
```

Comment:
This will have to move to after the compound initialization now, otherwise `event_source` will be overridden to `NULL` by the compound initialization.

Reply:
oops, didn't notice this :-(

Reply:
Nvm how you did it now is fine